jdaviz/conftest.py

```diff
@@ -279,6 +282,130 @@ def mos_image():
+)
+def create_wfi_image_model(image_shape, **kwargs):
+    """
+    Create a dummy Roman WFI Image datamodel instance with valid values
+    for attributes required by the schema.
+
+    Requires roman_datamodels >= 0.14.2.dev
+
+    Adapted from:
+        https://github.com/spacetelescope/roman_datamodels/blob/
+        932afed416e02bf0e684973f03548eeefe973b9e/src/roman_datamodels/testing/factories.py#L1224
+
+    Parameters
+    ----------
+    image_shape : tuple
+        Shape of the synthetic image to produce
+
+    **kwargs
+        Additional or overridden attributes.
+    Returns
+    -------
+    roman_datamodels.datamodel.ImageModel
+    """
+    from roman_datamodels import datamodels as rdd, stnode
+    from roman_datamodels.random_utils import (
+        generate_array_float32, generate_array_uint32, generate_array_uint16
+    )
+    from roman_datamodels.testing.factories import (
+        create_meta, create_cal_logs, create_photometry
+    )
+    count_rate_lims = dict(min=0, max=1e4)
+    raw = {
+        "data": generate_array_float32(
+            image_shape, units=u.electron / u.s, **count_rate_lims
+        ),
+        "dq": generate_array_uint32(image_shape),
+        "err": generate_array_float32(
+            image_shape, units=u.electron / u.s, **count_rate_lims
+        ),
+        "meta": create_meta(),
+        "var_flat": generate_array_float32(
+            image_shape, units=u.electron**2 / u.s**2, **count_rate_lims
+        ),
+        "var_poisson": generate_array_float32(
+            image_shape, units=u.electron**2 / u.s**2, **count_rate_lims
+        ),
+        "var_rnoise": generate_array_float32(
+            image_shape, units=u.electron**2 / u.s**2, **count_rate_lims
+        ),
+        "amp33": generate_array_uint16((2, image_shape[0] + 8, 128), units=u.DN),
+        "border_ref_pix_right": generate_array_float32(
+            (2, image_shape[0] + 8, 4), units=u.DN, **count_rate_lims
+        ),
+        "border_ref_pix_left": generate_array_float32(
+            (2, image_shape[0] + 8, 4), units=u.DN, **count_rate_lims
+        ),
+        "border_ref_pix_top": generate_array_float32(
+            (2, 4, image_shape[1] + 8), units=u.DN, **count_rate_lims
+        ),
+        "border_ref_pix_bottom": generate_array_float32(
+            (2, 4, image_shape[1] + 8), units=u.DN, **count_rate_lims
+        ),
+        "dq_border_ref_pix_right": generate_array_uint32((image_shape[0] + 8, 4)),
+        "dq_border_ref_pix_left": generate_array_uint32((image_shape[0] + 8, 4)),
+        "dq_border_ref_pix_top": generate_array_uint32((4, image_shape[1] + 8)),
+        "dq_border_ref_pix_bottom": generate_array_uint32((4, image_shape[1] + 8)),
+        "cal_logs": create_cal_logs(),
+    }
+    raw.update(kwargs)
+    raw["meta"]["photometry"] = create_photometry()
+    raw["meta"]["wcs"] = create_example_gwcs(image_shape)
```

Comment:
This line is the main difference from the analogous method in `roman_datamodels.testing.factories.create_wfi_image`. We need a synthetic GWCS object to use in our example image file, so I adapted a photutils method for creating one, and attach it to the image model.